java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -272,6 +274,10 @@ public Tree declaration() {
     return null;
   }
 
+  public void callbackOnceComplete(Runnable callback) {
```

Comment:
Not convinced by usage of `Runnable` here. We just want to execute an action. While to me the `Runnable` interface refers closely to thread handling, which are not relevant in this context.

Reply:
Runnable is the recommended functional interface for lambdas with no parameter nor return : see functional interface section in the documentation of runnable https://docs.oracle.com/javase/8/docs/api/java/lang/Runnable.html

Reply:
https://stackoverflow.com/a/40153253/1263543